static/js/publisher/release/actions/releases.js

```diff
@@ -1,5 +1,230 @@
+import "whatwg-fetch";
+
 export const UPDATE_RELEASES = "UPDATE_RELEASES";
 
+import { RISKS_WITH_AVAILABLE as RISKS } from "../constants";
+
+import { hideNotification, showNotification } from "./globalNotification";
+import { cancelPendingReleases } from "./pendingReleases";
+import { releaseRevisionSuccess, closeChannelSuccess } from "./channelMap";
+import { updateRevisions } from "./revisions";
+
+import { getRevisionsMap, initReleasesData } from "../releasesState";
+
+const ERROR_MESSAGE =
+  "There was an error while processing your request, please try again later.";
+
+function updateReleasesData(dispatch, releasesData) {
```

Comment:
This feels like it would fit into being action creator function as it technically just dispatches actions to update the data:
```
function updateReleasesData(releasesData) {
  return dispatch => {
    // init channel data in revisions list
    const revisionsMap = getRevisionsMap(releasesData.revisions);
    initReleasesData(revisionsMap, releasesData.releases);

    dispatch(updateRevisions(revisionsMap));
    dispatch(updateReleases(releasesData.releases));
  }
}
```

And of course when it's used it would be `dispatch(updateReleasesData)`. I guess in theory this action could be useful in other places as well.

Reply:
I've refactored this out - if there isn't a use case for it now, I figured we didn't need to make it too generic? I'll happily change it to what you suggested if you think it would be better this way though.